src/Factory/BlockFactory.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Setono\SyliusCMSPlugin\Factory;
+
+use Setono\SyliusCMSPlugin\Doctrine\ORM\BlockRepository;
+use Setono\SyliusCMSPlugin\Model\Block;
+use Setono\SyliusCMSPlugin\Model\BlockInterface;
+use Setono\SyliusCMSPlugin\Repository\BlockRepositoryInterface;
+use Sylius\Component\Resource\Factory\FactoryInterface;
+
+final class BlockFactory implements BlockFactoryInterface
+{
+    private BlockRepository $repository;
+
+    private FactoryInterface $decoratedFactory;
+
+    public function __construct(
+        FactoryInterface $decoratedFactory,
+        BlockRepositoryInterface $repository
+    ) {
+        $this->decoratedFactory = $decoratedFactory;
+        $this->repository = $repository;
+    }
+
+    public function createNew(): BlockInterface
+    {
+        return $this->decoratedFactory->createNew();
+    }
+
+    public function createNewFromResource(string $resourceId): BlockInterface
```

Comment:
```suggestion
    public function createNewFromResource($resource): BlockInterface
```

I think we should allow both int, string and actual resource to be injected here. That's what I would expect from the method name at least